sources/ClangSharp.PInvokeGenerator/PInvokeGenerator.cs

```diff
@@ -1922,6 +1922,14 @@ private Type[] GetBitfieldCount(RecordDecl recordDecl)
 
         private CallingConvention GetCallingConvention(Cursor cursor, Cursor context, Type type)
         {
+            if (cursor is FunctionDecl functionDecl)
+            {
+                if (functionDecl.IsVariadic)
+                {
+                    return CallingConvention.Cdecl;
+                }
+            }
```

Comment:
This needs to support remapping still. The default should definitely be `Cdecl`, but ClangSharp supports users doing any kind of remapping, even ones that technically may not make sense.

Reply:
The document of [`__stdcall`](https://docs.microsoft.com/en-us/cpp/cpp/stdcall) indicates that even the vararg function is marked as `__stdcall`, the compiler will makes it a `__cdecl`. Therefore I force `Cdecl` here.

Reply:
Yes, but that's not necessarily true for any and all potential platforms. ClangSharp has reasonable defaults and allows devs to override the default to fit their own needs.

Reply:
I think we are generating proper dll imports. If a programmer writes a method like
``` cpp
extern "C" int __stdcall foo(...);
```
The compiler won't raise errors. However, if we generate bindings like
``` csharp
[DllImport("foo.dll", CallingConvention = CallingConvention.StdCall)]
static extern int foo(__arglist);
```
The program will raise runtime `System.TypeLoadException` says that varargs function should use cdecl.